inst/tests/tests.Rraw

```diff
@@ -21426,3 +21426,27 @@ test(2330.7, as.data.table(list(z), keep.rownames=TRUE), data.table(rn=rep("", 3
 
 M <- matrix(1:6, nrow=3, dimnames=list(rep("", 3), c("V1", "V2")))   #  test of list(M) for empty-rowname'd matrix input
 test(2330.8, as.data.table(list(M), keep.rownames=TRUE), data.table(rn=rep("", 3), V1=1:3, V2=4:6))
+
+#2606 Recursive tables() naming convention
+test(2331.1, local({
```

Comment:
I would combine these three into one test:

```r
local({
  lst_named = list(...)
  lst_unnamed = list(...)
  nested = list(...)
  test(, tables(recursive=TRUE)$NAME, c(...))
})
```

Please also include tests that cover:

 - Mixed named/unnamed simple lists: `list(data.table(..), b=data.table(...))`
 - Mixing named/unnamed in a nested setting too